lib/connect/tls.js

```diff
@@ -20,7 +20,8 @@ function buildBuilder (mqttClient, opts) {
 
   function handleTLSerrors (err) {
     // How can I get verify this error is a tls error?
-    if (opts.rejectUnauthorized) {
+    // Only throw error if it is not an connection refused error to reconnect
+    if (opts.rejectUnauthorized && err && err.code !== 'ECONNREFUSED') {
```

Comment:
instead of checking if it is not an `'ECONNREFUSED'`, can we verify if it is a TLS error?